### ISSUE TITLE
Missing Values Crash ML

### ISSUE THREAD

**Issue by dstuck** [pgmpy/pgmpy#687]
I'm trying to get estimates for conditional probabilities for a completely observed graphical model with some missing data. I attempted to simply add the edges to the model and fit to my data in a pandas data frame, but it looks like the base estimator class attempts to coerce the columns to type int which crashes. Since in pandas integers with missing data (np.nan) are of type float and can't be cast to int I'm curious if this was not intended to work with missing data, but the MLE code seems to have checks for missing columns so I'm not sure if this handled missing data at some point or if I'm missing some way to handle the missing values differently.

Removing the cast to int in estimators/base.py allows the code to run but I'm not sure if this actually handling things properly in MLE or if there are other changes to make to get proper estimates.

**Comment by yashu-seth** [comment]
@dstuck Can you please share the exact steps so that we can reproduce the bug?

**Comment by dstuck** [comment]
Sorry about that. Here's a basic case that will break. Removing the astype(int) in base.py allows it to run, but I'm not 100% sure it handles the missing data properly.

``` python
import pgmpy
import pandas as pd
test_model = pgmpy.models.BayesianModel([('A', 'B'), ('B', 'C')])
test_data = pd.DataFrame(
    {'A':[0, 1, np.nan, np.nan], 'B':[0, 1, 0, 1], 'C':[np.nan, np.nan, 0, 1]}
)
test_model.fit(test_data)
```

**Comment by ankurankan** [comment]
@dstuck I don't think the current code handles the case of missing values.

**Comment by chrisittner** [comment]
@ankurankan This can be closed, fixed in #694:

``` python
import pgmpy
import pandas as pd
import numpy as np
test_model = pgmpy.models.BayesianModel([('A', 'B'), ('B', 'C')])
test_data = pd.DataFrame(
    {'A':[0, 1, np.nan, np.nan], 'B':[0, 1, 0, 1], 'C':[np.nan, np.nan, 0, 1]}
)
test_model.fit(test_data)

for cpd in test_model.get_cpds():
    print(cpd)
```

Output:

```
╒════════╤═════╕
│ A(0.0) │ 0.5 │
├────────┼─────┤
│ A(1.0) │ 0.5 │
╘════════╧═════╛
╒══════╤════════╤════════╕
│ A    │ A(0.0) │ A(1.0) │
├──────┼────────┼────────┤
│ B(0) │ 0.5    │ 0.5    │
├──────┼────────┼────────┤
│ B(1) │ 0.5    │ 0.5    │
╘══════╧════════╧════════╛
╒════════╤══════╤══════╕
│ B      │ B(0) │ B(1) │
├────────┼──────┼──────┤
│ C(0.0) │ 0.5  │ 0.5  │
├────────┼──────┼──────┤
│ C(1.0) │ 0.5  │ 0.5  │
╘════════╧══════╧══════╛
```
